packages/fast-layouts-react/src/horizontal-overflow/horizontal-overflow.tsx

```diff
@@ -0,0 +1,311 @@
+        }
+    }
+};
+
+export interface IHorizontalOverflowState {
+    items: HTMLElement[];
+    itemsHeight: string;
+}
+
+class HorizontalOverflow extends Foundation<HorizontalOverflowProps, IHorizontalOverflowState> {
+
+    public static getDerivedStateFromProps(
+        nextProps: HorizontalOverflowProps,
+        prevState: IHorizontalOverflowState
+    ): null | Partial<IHorizontalOverflowState> {
+        const items: HTMLElement[] = Array.prototype.slice.call(getItems(nextProps));
+
+        if (items !== prevState.items) {
+            return {
+                items
+            };
+        }
+    }
+
+    private horizontalOverflowItemsRef: React.RefObject<HTMLDivElement>;
+
+    constructor(props: HorizontalOverflowProps) {
+        super(props);
+
+        this.horizontalOverflowItemsRef = React.createRef();
+
+        this.state = {
+            items: [],
+            itemsHeight: "0"
+        };
+    }
+
+    /**
+     * Renders the Horizontal Overflow markup
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div
+                className={this.props.managedClasses.horizontalOverflow}
+                onLoad={this.itemsOnLoad}
+                style={{height: this.state.itemsHeight}}
+            >
+                <div className={this.props.managedClasses.horizontalOverflow_previous}>
+                    <div>
+                        <div onClick={this.handlePreviousClick}>
+                            {this.getPrevious()}
+                        </div>
+                    </div>
+                </div>
+                <div
+                    className={this.props.managedClasses.horizontalOverflow_items}
+                    ref={this.horizontalOverflowItemsRef}
+                >
+                    {this.getArrayItems()}
+                </div>
+                <div className={this.props.managedClasses.horizontalOverflow_next}>
+                    <div>
+                        <div onClick={this.handleNextClick}>
+                            {this.getNext()}
+                        </div>
+                    </div>
+                </div>
+            </div>
+        );
+    }
+
+    public componentDidMount(): void {
+        if (getItems(this.props).length === 0) {
+            return;
+        }
+
+        const items: HTMLElement[] = Array.prototype.slice.call(getItems(this.props));
+        const itemsHeight: string = this.getItemMaxHeight();
+
+        this.setState({
+            items,
+            itemsHeight
+        });
+    }
+
+    private itemsOnLoad = (): void => {
+        const itemsHeight: string = this.getItemMaxHeight();
+
+        if (itemsHeight !== this.state.itemsHeight) {
+            this.setState({
+                itemsHeight
+            });
+        }
+    }
+
+    private getItemMaxHeight(): string {
+        let itemMaxHeight: number = 0;
+
+        if (get(this.horizontalOverflowItemsRef, "current.childNodes")) {
```

Comment:
You're accessing `get(this.horizontalOverflowItemsRef, "current.childNodes")` twice here - you should store that in a variable to avoid that.